llvm/lib/Transforms/Scalar/Scalarizer.cpp

```diff
@@ -727,7 +727,8 @@ bool ScalarizerVisitor::splitCall(CallInst &CI) {
 
   SmallVector<llvm::Type *, 3> Tys;
   // Add return type if intrinsic is overloaded on it.
-  if (isVectorIntrinsicWithOverloadTypeAtArg(ID, -1))
+  if (TTI->isVectorIntrinsicWithOverloadTypeAtArg(
```

Comment:
Passing in a default is a weird code pattern. also if you don't pass a target looking at TTI seems wrong. 
just do `isVectorIntrinsicWithOverloadTypeAtArg... && TTI->isVectorIntrinsicWithOverloadTypeAtArg...`

Reply:
I agree, the problem is that the default behaviour is to only return `true` for `OpScalarIdx == -1`. Which in the `asdouble` case we don't want to add the `double` return type, so `||` doesn't work and when we want to add any other different index `&&` won't work.

Reply:
I see. Still don't like the default behavior though. How about instead Create a helper in this file.  and check `Intrinsic::isTargetIntrinsic(ID)` if it is true do `TTI->isVectorIntrinsicWithOverloadTypeAtArg`. if it is false do `isVectorIntrinsicWithOverloadTypeAtArg`.

That way TTI isn't check unless we know its a target intrinsic.